### PR TITLE
feat(CI): Run different flow according to version

### LINKED ISSUE #472: trigger corresponding test workflow
for now, we have test workflow for v1alpha2 and v1alpha1, v1alpha1 test workflow should be triggered only when v1alpha1 code changed, so does v1alpha2

### PR DESCRIPTION
**Pull request by gaocegege**
Signed-off-by: Ce Gao <gaoce@caicloud.io>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #472 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/516)

### CODE REVIEW

File: prow_config.yaml [hunk 1]
```diff
@@ -19,9 +39,29 @@ workflows:
...
+    - pkg/controller/v1alpha1/*
+    - pkg/db/v1alpha1/*
+    - pkg/earlystopping/v1alpha1/*
+    - pkg/manager/v1alpha1/*
+    - pkg/manager/modelstore/*
+    - pkg/suggestion/v1alpha1/*
+    - pkg/ui/*
+    - cmd/earlystopping/*
+    - cmd/katib-controller/v1alpha1/*
+    - cmd/manager/v1alpha1/*
```
Comment (hougangliu) [inline prow_config.yaml]: both v1alpha1 and v1alpha2 share same Dockerfile for some sub dir, we should include the Dockerfile in two workflows

Comment (gaocegege) [reply prow_config.yaml]: Make sense, addressed. Thanks

**Pull request merged**